common/src/main/java/org/terracotta/angela/common/util/RetryUtils.java

```diff
@@ -41,20 +42,24 @@
    * @return {@code true} if the condition was evaluated to true within the given constraints, false otherwise
    */
   public static boolean waitFor(Callable<Boolean> condition, int maxWaitTimeMillis) {
-    int currRetryTime = min(maxWaitTimeMillis, 100);
-    int timeRemaining = maxWaitTimeMillis;
+    long startTime = currentTimeMillis();
+    long timeElapsed = 0;
+    long currRetryTime = min(maxWaitTimeMillis, 100);
     ExecutorService executorService = Executors.newSingleThreadExecutor(Thread::new);
     boolean success = false;
 
     while (currRetryTime <= maxWaitTimeMillis) {
-      long timeout = min(currRetryTime, timeRemaining); //Keep track of how much time remains
+      long timeout = min(currRetryTime, maxWaitTimeMillis - timeElapsed); //Keep track of how much time remains
```

Comment:
why min ? if we say min, then most of the time it will end up being 100, and at the very end, it could be the remaining of `axWaitTimeMillis - timeElapsed` which could even be negative...

Reply:
`currRetryTime` is being incremented later in the `else` part. So, the min shouldn't be 100.

About the `maxWaitTimeMillis - timeElapsed` being negative. That could happen. I can add the check in the `while` condition.